File: structural/dart/flyweight_structured/interfaces.dart
// Flyweight Interface - defines the operations that can be performed on flyweight objects
abstract class TreeType {
  void draw(int x, int y);
  String get name;
  String get color;
}

abstract class CharacterFormat {
  void apply(String text);
  String get font;
  int get size;
  String get color;
}

abstract class SoldierType {
  void render(int x, int y, String weapon);
  String get uniform;
  int get speed;
}

File: structural/typescript/flyweight_structured/interfaces.ts
// Flyweight Interface - defines the operations that can be performed on flyweight objects
export interface TreeType {
  draw(x: number, y: number): void;
  getName(): string;
  getColor(): string;
}

export interface CharacterFormat {
  apply(text: string): void;
  getFont(): string;
  getSize(): number;
  getColor(): string;
}

export interface SoldierType {
  render(x: number, y: number, weapon: string): void;
  getUniform(): string;
  getSpeed(): number;
}
